src/main/java/com/yahoo/bullet/parsing/BinaryExpression.java

```diff
@@ -0,0 +1,55 @@
+/*
+ *  Copyright 2018, Oath Inc.
+ *  Licensed under the terms of the Apache License, Version 2.0.
+ *  See the LICENSE file associated with the project for terms.
+ */
+package com.yahoo.bullet.parsing;
+
+import com.google.gson.annotations.Expose;
+import com.yahoo.bullet.common.BulletError;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static com.yahoo.bullet.common.BulletError.makeError;
+
+@Getter @Setter
+public class BinaryExpression extends Expression {
+    public static final BulletError BINARY_EXPRESSION_REQUIRES_VALID_EXPRESSIONS_ERROR =
+            makeError("This Expression needs 2 valid operands", "Please add an expression for 'left' and 'right'.");
+
+    @Expose
+    private Expression left;
+    @Expose
+    private Expression right;
+
+    /**
+     * Default Constructor. GSON recommended.
+     */
+    public BinaryExpression() {
+        super();
+        left = null;
+        right = null;
+    }
+
+    @Override
+    public String toString() {
+        return "{" + super.toString() + ", left: " + left + ", right: " + right + "}";
+    }
+
+    @Override
+    public Optional<List<BulletError>> initialize() {
+        if (left == null || right == null) {
+            return Optional.of(Collections.singletonList(BINARY_EXPRESSION_REQUIRES_VALID_EXPRESSIONS_ERROR));
```

Comment:
If we do this, then we should have two strings templated out and construct it instead of hardcoding. We do this for a couple of other errors but then again we don't for others.

Reply:
We can do a follow up PR to change all of them if needed.